assets/js/data/collections/controls.js

```diff
@@ -26,15 +26,15 @@ export const apiFetchWithHeaders = ( path ) => {
  */
 export const controls = {
 	API_FETCH_WITH_HEADERS( { path } ) {
-		return new Promise( ( resolve, reject ) => {
+		return new Promise( ( resolve ) => {
 			triggerFetch( { path, parse: false } )
 				.then( ( response ) => {
 					response.json().then( ( items ) => {
 						resolve( { items, headers: response.headers } );
 					} );
 				} )
 				.catch( ( error ) => {
-					reject( error );
+					throw error;
```

Comment:
@nerrad I started looking at this today and I'm trying to work everything out here, but one think I did see was that this exception is never caught, even with the catch you added in resolvers.js. Any clues as to why that might be the case?

Rather than throwing an exception here, would resolving and passing back some API details here be a good idea, so that it can be handled without an exception?

Reply:
> but one think I did see was that this exception is never caught, even with the catch you added in resolvers.js

🤔 hmm, that definitely doesn't match the description from when I created this pull. I'm not sure why it isn't working offhand yet, I'll get back to you when I spin up this pull and test.  I have a feeling with this pull being a draft, I wasn't complete on this yet (it was a work in progress), and I didn't get a chance to update the pull description when I moved it out of scope for the first cycle.

> Rather than throwing an exception here, would resolving and passing back some API details here be a good idea, so that it can be handled without an exception?

Where would you be expecting the details to be passed back to, the store?  Calling code? If the latter, calling code doesn't have access to the returned value from the resolver (hence using the store to keep the error state). Seeing this definitely makes me think this pull isn't complete yet.

Reply:
Ok thanks. To test this I hardcoded an error in the product API, and tried logging to console in the catch (resolvers.js). That never triggered and instead I saw an uncaught exception object in console which I was not able to trace.

Reply:
So what I had in mind was in API_FETCH_WITH_HEADERS rather than throw an exception again (which isn't being caught for whatever reason), we resolve with: `resolve( { items: [], headers: [], error } );`. Then the resolver can handle that if it's not undefined without using try catch itself.

Reply:
Throwing that error from `useCollection` also seems to be giving bad results as you end up with the `useCollection is not defined` error instead of the thrown error (I assume because useCollection is being exported). So I'm also trying from there to return the error - it can then be picked up and thrown by the caller.

Reply:
NVM, it seems ErrorBoundry still shows console errors so it may just be that :) Working on a new PR with some of my changes.

Reply:
> NVM, it seems ErrorBoundry still shows console errors

Oh, yes sorry, that should have been something I clarified.